src/peft/peft_model.py

```diff
@@ -397,9 +399,8 @@ def load_adapter(self, model_id, adapter_name, is_trainable=False, **kwargs):
                     f"Please check that the file {WEIGHTS_NAME} is present at {model_id}."
                 )
 
-        adapters_weights = torch.load(
-            filename, map_location=torch.device("cuda" if torch.cuda.is_available() else "cpu")
-        )
+        adapters_weights = safe_load_file(filename)
```

Comment:
You can have the device here directly too.

Reply:
When adding the device strangely I get: 
```bash
src/peft/peft_model.py:416: in load_adapter
    adapters_weights = safe_load_file(filename, device=torch.device("cuda" if torch.cuda.is_available() else "cpu"))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

filename = '/var/tmp/tmpwn39lu1x/adapter_model.safetensors', device = device(type='cuda')

    def load_file(filename: Union[str, os.PathLike], device="cpu") -> Dict[str, torch.Tensor]:
        """
        Loads a safetensors file into torch format.
    
        Args:
            filename (`str`, or `os.PathLike`)):
                The name of the file which contains the tensors
            device (`Dict[str, any]`, *optional*, defaults to `cpu`):
                The device where the tensors need to be located after load.
                available options are all regular torch device locations
    
        Returns:
            `Dict[str, torch.Tensor]`: dictionary that contains name as key, value as `torch.Tensor`
    
        Example:
    
        ```python
        from safetensors.torch import load_file
    
        file_path = "./my_folder/bert.safetensors"
        loaded = load_file(file_path)
        ```
        """
        result = {}
>       with safe_open(filename, framework="pt", device=device) as f:
E       safetensors_rust.SafetensorError: device cuda is invalid

../../../miniconda3/envs/fix-test/lib/python3.9/site-packages/safetensors/torch.py:259: SafetensorError
```
I am unsure what is missing here 😢